From: Markus Lehtonen <markus.lehtonen@linux.intel.com>
Date: Mon, 3 Sep 2012 15:21:43 +0300
Subject: Fedora HACK: use custom (older) xdelta

Signed-off-by: Markus Lehtonen <markus.lehtonen@linux.intel.com>
---
 pristine-gz  |    4 ++--
 pristine-tar |    4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/pristine-gz b/pristine-gz
index 3668256..4818e28 100755
--- a/pristine-gz
+++ b/pristine-gz
@@ -234,7 +234,7 @@ sub reproducegz {
 		else {
 			# generate a binary delta and see if this is the
 			# best variant so far
-			my $ret=system("xdelta delta -0 --pristine $tempout $orig $tempdir/tmpdelta 2>/dev/null") >> 8;
+			my $ret=system("xdelta1 delta -0 --pristine $tempout $orig $tempdir/tmpdelta 2>/dev/null") >> 8;
 			# xdelta exits 1 on success
 			if ($ret == 1) {
 				my $size=(stat("$tempdir/tmpdelta"))[7];
@@ -297,7 +297,7 @@ sub gengz {
 		my $tempdir=tempdir();
 		my $tfile="$tempdir/".basename($file).".gz";
 		doit_redir($file, $tfile, @zgz);
-		doit("xdelta", "patch", "--pristine", $delta->{delta}, $tfile, "$file.gz");
+		doit("xdelta1", "patch", "--pristine", $delta->{delta}, $tfile, "$file.gz");
 	}
 	else {
 		doit_redir("$file", "$file.gz", @zgz);
diff --git a/pristine-tar b/pristine-tar
index c75e4fc..b08702b 100755
--- a/pristine-tar
+++ b/pristine-tar
@@ -391,7 +391,7 @@ sub gentar {
 	my $ok;
 	foreach my $variant (@try) {
 		my $recreatetarball=$variant->();
-		my $ret=try_doit("xdelta", "patch", $delta->{delta}, $recreatetarball, $out);
+		my $ret=try_doit("xdelta1", "patch", $delta->{delta}, $recreatetarball, $out);
 		if ($ret == 0) {
 			$ok=1;
 			last;
@@ -503,7 +503,7 @@ sub gendelta {
 	}
 
 	$delta{delta}="$tempdir/delta";
-	my $ret=system("xdelta delta -0 --pristine $recreatetarball $tarball $delta{delta}") >> 8;
+	my $ret=system("xdelta1 delta -0 --pristine $recreatetarball $tarball $delta{delta}") >> 8;
 	# xdelta exits 1 on success if there were differences
 	if ($ret != 1 && $ret != 0) {
 		error "xdelta failed with return code $ret";
-- 
1.7.7

